CMakeLists.txt

```diff
@@ -17,38 +17,88 @@ message("-- Version: ${VERSION}")
 set(GENERIC_LIB_VERSION ${VERSION})
 string(SUBSTRING ${VERSION} 0 1 GENERIC_LIB_SOVERSION)
 
-# Try and enable C++11. Don't use C++14 because it doesn't work in some
-# configurations.
+# Import our CMake modules
 include(CheckCXXCompilerFlag)
 include(AddCXXCompilerFlag)
 include(CXXFeatureCheck)
 
-check_cxx_compiler_flag(-std=c++11 HAVE_FLAG_CXX_11)
-check_cxx_compiler_flag(-std=c++0x HAVE_FLAG_CXX_0X)
-if (HAVE_FLAG_CXX_11)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-elseif (HAVE_FLAG_CXX_0X)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+# Determine the C++ standard to use
+# The C and C++ standards to use
+set(BENCHMARK_CXX_STANDARD auto CACHE STRING
+  "The C++ standard to use; auto uses the latest supported by the compiler")
+set_property(CACHE BENCHMARK_CXX_STANDARD PROPERTY STRINGS
+  auto c++14 c++11 c++98)
+if ("${BENCHMARK_CXX_STANDARD}" STREQUAL "auto")
+  # Try and enable C++11. Don't use C++14 because it doesn't work in some
+  # configurations.
+  #add_cxx_compiler_flag(-std=c++14)
+  if (NOT HAVE_CXX_FLAG_STD_CXX14)
```

Comment:
Yeah I think that makes sense. C++14 enables C11 which removes `gets(...)`. There are certain compiler/stdlib/libc combinations on linux where `gets(...)` is used but is not available in the headers. 

`check_cxx_compiler_flag(-std=c++14)` doesn't see this issue because the compile test does not include any system headers that use `gets(...)`.

Since we don't *need* any C++14 features it is easier to just limit the library to C++11 for now.